### PR TITLE
Plugin double prefix

### PR DESCRIPTION
**Pull request by mcollina**
This is an alternative to #1225 that creates much less breakage. Basically I'm registering two routes, one with and one without the ending slash. I'm up for adding a flag that have the more strict behavior that #1225 defines, if you think it's needed.

```js
'use strict'

const fastify = require('fastify')()

fastify.register(function (instance, options, next) {
  // the route will be called for both '/english/' and '/english'
  instance.get('/', (req, reply) => {
    reply.send({ greet: 'hello' })
  })
  next()
}, { prefix: '/english' })

fastify.register(function (instance, options, next) {
  // the handler will be called for both '/italian/' and '/italian'
  instance.get('/', (req, reply) => {
    reply.send({ greet: 'ciao' })
  })
  next()
}, { prefix: '/italian' })

fastify.listen(8000, function (err) {
  if (err) {
    throw err
  }
  console.log(`server listening on ${fastify.server.address().port}`)
})
```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by cemremengu** [comment]
If I am getting it right, this PR should also remove the `ignoreTrailingSlash` option which becomes redundant.

**Comment by mcollina** [comment]
@cemremengu it doesn't. That would still apply to normal routes.

**Comment by cemremengu** [comment]
So prefixing will ignore trailing slash but _normal_ routes will require the option?

**Comment by mcollina** [comment]
> So prefixing will ignore trailing slash but normal routes will require the option?

Yes, exactly. Note that this is very similar to our current approach. In a more clear way: 

| ignoreTrailingSlash | prefix    | route | matches on master | matches on this PR |
| --- | --- | ------- | -------| --- | 
| `false` | `/hello` | /         | `/hello`                    | `/hello` `/hello/`                        |
| `true` | `/hello` | /         | `/hello` `/hello/`     | `/hello` `/hello/`                       |

This is needed because from a plugin there is currently no way to match `/hello/` (with a prefix of `/hello`).

I think that would match what most users would expect, considering what @delvedor and @davidmarkclements said.

**Comment by mcollina** [comment]
I've added another test to clarify the behavior.

PTAL.

**Comment by mcollina** [comment]
Added two more tests for the `ignoreTrailingSlash` option.

Do you think we should document this behavior? If so, where?

**Comment by delvedor** [comment]
It should definitely be documented. What about [here](https://github.com/fastify/fastify/blob/master/docs/Routes.md#route-prefixing)?

**Comment by mcollina** [comment]
PTAL, have a look.

File: test/route-prefix.test.js [hunk 1]
```diff
@@ -333,3 +333,123 @@ test('Can retrieve prefix within encapsulated instances', t => {
...
+  }, (err, res) => {
+    t.error(err)
+    t.same(JSON.parse(res.payload), { hello: 'world' })
+  })
+})
+
+test('matches both /prefix and /prefix/ with a / route - ignoreTrailingSlash: false', t => {
+  t.plan(4)
+  const fastify = Fastify({
+    ignoreTrailingSlash: true
```
Comment (nwoltman) [inline test/route-prefix.test.js]: The test name says `ignoreTrailingSlash: false`, so I'm guessing this should be `false`.

**Comment by mcollina** [comment]
@nwoltman done, good catch!

**Pull request merged**